### PR TITLE
Automate release

### PR DESCRIPTION
**Pull request by jkleinsc**
New scripts to automate our release process.

### CODE REVIEW

**Comment by jkleinsc** [comment]
@zeke yes

File: script/release.js [hunk 1]
```diff
@@ -0,0 +1,392 @@
...
+
+async function validateReleaseAssets (draft) {
+  const requiredAssets = assetsForVersion(draft.tag_name).sort()
+  const extantAssets = draft.assets.map(asset => asset.name).sort()
+  const downloadUrls = draft.assets.map(asset => asset.browser_download_url).sort()
+
+  requiredAssets.forEach(asset => {
+    check(extantAssets.includes(asset), asset)
+  })
+  check((failureCount === 0), `All required GitHub assets exist for release`, true)
```
Comment (vanessayuenn) [inline script/release.js]: In the case where one of the checks before this one has failed, but the process hasn't exited, you would get an inaccurate error message saying `All required GitHub assets exist for release` even if the assets are all there.

File: script/release.js [hunk 2]
```diff
@@ -0,0 +1,392 @@
...
+    'ffmpeg-{{VERSION}}-linux-arm64.zip',
+    'ffmpeg-{{VERSION}}-linux-armv7l.zip',
+    'ffmpeg-{{VERSION}}-linux-ia32.zip',
+    'ffmpeg-{{VERSION}}-linux-x64.zip',
+    'ffmpeg-{{VERSION}}-mas-x64.zip',
+    'ffmpeg-{{VERSION}}-win32-ia32.zip',
+    'ffmpeg-{{VERSION}}-win32-x64.zip',
+    'SHASUMS256.txt'
+  ]
+  return patterns.map(pattern => pattern.replace(/{{VERSION}}/g, version))
```
Comment (vanessayuenn) [inline script/release.js]: Not a huge deal but you can consider omitting this replace operation and use template literals within the array.

File: script/prepare-release.js [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+    console.log(`${pass} Successfully pushed the release branch.  Wait for ` +
+      `release builds to finish before running "npm run release".`)
+  } else {
+    console.log(`${fail} Error pushing the release branch: ` +
+        `${pushDetails.stderr}`)
+    process.exit(1)
+  }
+}
+
+async function prepareRelease (isBeta, notesOnly) {
```
Comment (vanessayuenn) [inline script/prepare-release.js]: `isBeta` isn't used in the function

**Comment by jkleinsc** [comment]
@zeke @vanessayuenn I updated this PR to address the @vanessayuenn's concerns as well as improvements/changes needed once I used it to run through the release process.  I'd appreciate your 👀  on it.  Thanks!

File: docs/development/releasing.md [hunk 4]
```diff
@@ -2,47 +2,51 @@
...
 This document describes the process for releasing a new version of Electron.
 
-## Find out what version change is needed
-Is this a major, minor, patch, or beta version change? Read the [Version Change Rules](../tutorial/electron-versioning.md#version-change-rules) to find out.
-
-## Create a temporary branch
+## Determine which branch to release from
 
 - **If releasing beta,** create a new branch from `master`.
 - **If releasing a stable version,** create a new branch from the beta branch you're stablizing.
```
Comment (zeke) [inline docs/development/releasing.md]: typo: stablizing

File: docs/development/releasing.md [hunk 5]
```diff
@@ -159,65 +163,46 @@ This release is published to [npm](https://www.npmjs.com/package/electron) under
...
 1. Wait for all builds to pass before proceeding.
+1. You can run `npm run release --validateRelease` to verify that all of the
+required files have been created for the release.
 
 ## Merge temporary branch
-
-Merge the temporary branch back into master, without creating a merge commit:
-
+Once the release builds have finished, merge the `release` branch back into
+master using the `merge-release` script.
```
Comment (zeke) [inline docs/development/releasing.md]: `back into master` (or the source release branch) right?

File: script/merge-release.js [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+async function mergeRelease () {
+  await checkoutBranch(branchToRelease)
+  let mergeSuccess = await mergeReleaseIntoBranch(branchToRelease)
+  if (mergeSuccess) {
+    console.log(`${pass} Successfully merged release branch into ` +
+      `${branchToRelease}.`)
+    await commitMerge()
+    let pushSuccess = await pushBranch(branchToRelease)
+    if (pushSuccess) {
+      console.log(`${pass} Success!!! ${branchToRelease} now has the latest release!`)
```
Comment (zeke) [inline script/merge-release.js]: 🎉

File: script/prepare-release.js [hunk 7]
```diff
@@ -0,0 +1,173 @@
...
+    console.log(`${fail} Aborting because draft release for
+      ${drafts[0].release.tag_name} already exists.`)
+    process.exit(1)
+  }
+  console.log(`${pass} A draft release does not exist; creating one.`)
+  githubOpts.body = releaseNotes
+  githubOpts.draft = true
+  githubOpts.name = `electron ${newVersion}`
+  if (isBeta) {
+    githubOpts.body = `Note: This is a beta release.  Please file new issues ` +
```
Comment (zeke) [inline script/prepare-release.js]: I'm okay with the multiline concatenation, but FYI [dedent](http://ghub.io/dedent) is a nice module for this.

**Comment by zeke** [comment]
✨ 🤖 ✨

**Pull request merged**